decompressed/base/etc/init.d/rootdevice

```diff
@@ -252,56 +229,95 @@ install_gui() {
-        uci set modgui.var.bank_check="$get_config_bank1"
+  if [[ "$bootedbank" != "$activebank" && "$bootedbank" == "bank_2" ]]; then
+    #TODO: add checks to be sure the bank_1 is unbootable (is this really needed?)
+    #TODO: add checks to be sure the bank_1 overlay is SAFE and bootable (empty/prerooted?)
+    echo "  OBP sanity check succeeded continue boot complete..." >/dev/console
+  else
+    #OBP not OK, check if was an user choice otherwise automatically apply (this should be the case called also on first install)
+
+    if [ ! "$(uci -q get modgui.var.check_obp)" ]; then
+      #we must inizialize the check_obp var as is not found
+      if [ -f $notbooted_modgui_configfile ]; then
+        #if we are in a clean bank_2 overlay but in the notbooted bank overlay there was a modgui config file restore var from it
+        get_config_notbooted=$(< $notbooted_modgui_configfile grep check_obp | awk '{print $3}' | sed "s/'//g")
+        if [ -n "$get_config_notbooted" ] && [ ! "$(uci -q get modgui.var.check_obp)" ]; then
+          uci set modgui.var.check_obp="$get_config_notbooted"
+          uci commit modgui
+        fi
+      else
+        #inizialize check_obp default value and only if cannot read the var from elsewhere (inactive bank config)
+        uci set modgui.var.check_obp="1"
         uci commit modgui
       fi
-    else
-      uci set modgui.var.bank_check="1"
-      uci commit modgui
     fi
-    if [ "$(uci -q get modgui.var.bank_check)" == "0" ]; then
-      logger_command "Firmware in Bank_2 but check_bank set to 0... Not transferring"
-      echo "  Firmware in Bank_2 but check_bank set to 0... Not transferring" >/dev/console
+
+    if [ "$(uci -q get modgui.var.check_obp)" == "0" ]; then
+      logger_command "OBP sanity check FAILED but check_obp set to 0... Not acting"
+      echo "  OBP sanity check FAILED but check_obp set to 0... Not acting" >/dev/console
     else
-      logger_command "Firmware in Bank_2... Transferring"
-      echo "  Firmware in Bank_2... Transferring" >/dev/console
-      transfer_bank
+      logger_command "OBP sanity check FAILED start autoplanning..."
+      echo "  OBP sanity check FAILED start autoplanning..." >/dev/console
+
+      if [ "$bootedbank" == "bank_2" ]; then
+        # Activate bank_1
+        echo bank_1 > /proc/banktable/active
+        # Erase firmware in bank_1
+        mtd erase bank_1
+      else
+        # Make a temp copy of the firmware in bank_1
+        dd if=/dev/mtd3 of=/tmp/bank.fw
+        # Flash that copy into bank_2
+        mtd write /tmp/bank.fw bank_2
+        # Clean temp firmware copy
+        rm /tmp/bank.fw
+        # Clean any existing overlay for bank_2 firmware
+        rm -rf /overlay/bank_2
+        # Make a temp copy of overlay for bank_1 firmware
+        cp -rf /overlay/bank_1 /tmp/bank_overlay_backup
+        # Free up overlay space by removing existing overlay for bank_1 firmware
```

Comment:
perchè mettee in mezzo `bank_overlay_backup`? non possiamo fare un più pulito `mv /overlay/bank_1 /overlay/bank_2` ?

Reply:
Non mi ricordo perchè avevo evitato il mv, forse c'era un motivo risalente a versioni precedenti di quei comandi

Reply:
Il timore è sempre lo stesso degli OOM...

Reply:
Di solito a me preoccupano più gli oom sul'overlay, e questo è di sicuro il motivo per cui non gli fatto la semplice copia. Però il motivo del passaggio in memoria qui non me lo ricordo proprio, mi ricordo che era intenzionale ma non ricordo qual era.

Reply:
in che senso gli oom sull'overlay? qui se ne capita uno capiterebbe mentre riempe /tmp, l'unico rischio è che si ripeta poi ad ogni boot senza mai riuscire a terminare

con l'`mv` speravo si riuscisse a fare tutto senza toccare quasi per nulla la RAM e teoricamente con un unica operazione atomica